lib/services/kubernetes.go

```diff
@@ -220,7 +222,44 @@ func labelsFromAzureKubeCluster(cluster *azure.AKSCluster) map[string]string {
 	return labels
 }
 
-// NewKubeClusterFromAWSEKS creates a database resource from an EKS cluster.
+// NewKubeClusterFromGCPGKE creates a kube_cluster resource from an GKE cluster.
+func NewKubeClusterFromGCPGKE(cluster gcp.GKECluster) (types.KubeCluster, error) {
+	return types.NewKubernetesClusterV3(
+		setKubeName(types.Metadata{
+			Description: getOrSetDefaultGCPDescription(cluster),
+			Labels:      labelsFromGCPKubeCluster(cluster),
+		}, cluster.Name),
+		types.KubernetesClusterSpecV3{
+			GCP: types.KubeGCP{
+				Name:      cluster.Name,
+				ProjectID: cluster.ProjectID,
+				Location:  cluster.Location,
+			},
+		})
+}
+
+// getOrSetDefaultGCPDescription gets the default GKE cluster description if available,
+// otherwise returns a default one.
+func getOrSetDefaultGCPDescription(cluster gcp.GKECluster) string {
+	if len(cluster.Description) > 0 {
+		return cluster.Description
+	}
+	return fmt.Sprintf("GCP GKE cluster %q in %s",
+		cluster.Name,
+		cluster.Location)
+}
+
+// labelsFromGCPKubeCluster creates kube cluster labels.
+func labelsFromGCPKubeCluster(cluster gcp.GKECluster) map[string]string {
+	labels := maps.Clone(cluster.Labels)
+	labels[types.OriginLabel] = types.OriginCloud
```

Comment:
I wonder if we should add a system "teleport.dev/cloud" label to indicate where it was imported from. Maybe not in this PR though.